src/main/java/com/google/devtools/build/remote/client/RemoteClient.java

```diff
@@ -118,14 +135,130 @@ private void listOutputDirectory(OutputDirectory dir, int limit) throws IOExcept
   }
 
   // Recursively list directory files/subdirectories with digests given a Tree of the directory.
-  private void listTree(Path path, Tree tree, int limit) throws IOException, InterruptedException {
+  private void listTree(Path path, Tree tree, int limit) throws IOException {
     Map<Digest, Directory> childrenMap = new HashMap<>();
     for (Directory child : tree.getChildrenList()) {
       childrenMap.put(digestUtil.compute(child), child);
     }
     listDirectory(path, tree.getRoot(), childrenMap, limit);
   }
 
+  private static int getNumFiles(Tree tree) {
+    return tree.getChildrenList().stream().mapToInt(dir -> dir.getFilesCount()).sum();
```

Comment:
Nice! I just learned this syntax a few days ago, it's really cool! Java getting closer and closer to Python :-)